core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/conditional/CaseExpression.kt

```diff
@@ -0,0 +1,143 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.conditional
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.Resolvable
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.UnaliasedExpression
+import ch.ergon.dope.validtype.BooleanType
+import ch.ergon.dope.validtype.ValidType
+
+private const val CASE = "CASE"
+private const val WHEN = "WHEN"
+private const val THEN = "THEN"
+private const val END = "END"
+private const val ELSE = "ELSE"
+
+class CaseClass<T : ValidType>(val case: TypeExpression<T>) : Resolvable {
+    override fun toDopeQuery(): DopeQuery {
+        val caseDopeQuery = case.toDopeQuery()
+        return DopeQuery("$CASE ${caseDopeQuery.queryString}", caseDopeQuery.parameters)
+    }
+}
+
+class SimpleCaseExpression<T : ValidType, U : ValidType>(
+    val case: TypeExpression<T>,
+    val firstWhenThen: SearchResult<T, out U>,
```

Comment:
I think this is not very understandable. What is a "whenThen"? And it is a "SearchResult"?

Reply:
I renamed it to `...SearchResult`